resources/scripts/ttx_diff.py

```diff
@@ -314,6 +317,76 @@ def reduce_diff_noise(build_dir, fontc, fontmake):
         build_dir, fontc, fontmake, "gvar/glyphVariations", "glyph", "/tuple/delta"
     )
 
+# returns a dictionary of {"compiler_name":  {"tag": "xml_text"}}
+def generate_output(build_dir: Path, fontmake_ttf: Path, fontc_ttf: Path):
+    fontc_ttx = ttx(fontc_ttf)
+    fontmake_ttx = ttx(fontmake_ttf)
+    fontc_gpos = simple_gpos_output(fontc_ttf, build_dir / "fontc.markkern.txt")
+    fontmake_gpos = simple_gpos_output(
+        fontmake_ttf, build_dir / "fontmake.markkern.txt"
+    )
+
+    fontc = etree.parse(fontc_ttx)
+    fontmake = etree.parse(fontmake_ttx)
+    reduce_diff_noise(build_dir, fontc, fontmake)
+
+    fontc = extract_comparables(fontc, build_dir, "fontc")
+    fontmake = extract_comparables(fontmake, build_dir, "fontmake")
+    fontc[MARK_KERN_NAME] = fontc_gpos
+    fontmake[MARK_KERN_NAME] = fontmake_gpos
+    return {"fontc": fontc, "fontmake": fontmake}
+
+def print_output(build_dir: Path, output: dict[str, dict[str, str]]):
+    fontc = output["fontc"]
+    fontmake = output["fontmake"]
+    print("COMPARISON")
+    t1 = set(fontc.keys())
+    t2 = set(fontmake.keys())
+    if t1 != t2:
+        if t1 - t2:
+            tags = ", ".join(f"'{t}'" for t in sorted(t1 - t2))
+            print(f"  Only fontc produced {tags}")
+
+        if t2 - t1:
+            tags = ", ".join(f"'{t}'" for t in sorted(t2 - t1))
+            print(f"  Only fontmake produced {tags}")
+
+
+    for tag in sorted(t1 & t2):
+        t1s = fontc[tag]
+        t2s = fontmake[tag]
+        if t1s == t2s:
+            print(f"  Identical '{tag}'")
+        else:
+            difference = diff_ratio(t1s, t2s)
+            p1 = build_dir / path_for_output_item(tag, "fontc")
+            p2 = build_dir / path_for_output_item(tag, "fontmake")
+            print(f"  DIFF '{tag}', {p1} {p2} ({difference:.1%})")
+
+
+# given the ttx for a font, return a map of tags -> xml text for each root table.
+# also writes the xml to individual files
+def extract_comparables(font_xml, build_dir: Path, compiler: str) -> dict[str, str]:
+    comparables = dict()
+    tables = {e.tag: e for e in font_xml.getroot()}
+    for tag in sorted(e.tag for e in font_xml.getroot()):
+        table_str = etree.tostring(tables[tag])
+        path = build_dir / f"{compiler}.{tag}.ttx"
+        path.write_bytes(table_str)
+        comparables[tag] = table_str
+
+    return comparables
+
+
+def diff_ratio(text1: str, text2: str):
```

Comment:
Suggest a return type annotation and a comment to indicate how to interpret the value

Reply:
(or just switching to use `ratio`)